spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/support/converter/NotificationRequestConverter.java

```diff
@@ -120,12 +123,9 @@ public Object getMessage() {
                 if (MessageAttributeDataTypes.STRING.equals(attributeType)) {
                     messageHeaders.put(attributeName, attributeValue);
                 } else if (attributeType.startsWith(MessageAttributeDataTypes.NUMBER)) {
-                    Object numberValue = getNumberValue(attributeType, attributeValue);
-                    if (numberValue != null) {
-                        messageHeaders.put(attributeName, numberValue);
-                    }
-                } else if (MessageAttributeDataTypes.BINARY.equals(attributeName)) {
-                    messageHeaders.put(attributeName, ByteBuffer.wrap(attributeType.getBytes()));
+                    messageHeaders.put(attributeName, getNumberValue(attributeType, attributeValue));
+                } else if (MessageAttributeDataTypes.BINARY.equals(attributeType)) {
+                    messageHeaders.put(attributeName, ByteBuffer.wrap(attributeValue.getBytes()));
```

Comment:
Whilst going through this code, I noticed something peculiar: `MessageAttributeDataTypes.BINARY` was being compared with `attributeName` and the value was taken from `attributeType`.